subcommand/common/common.go

```diff
@@ -46,3 +51,48 @@ func ValidatePort(flagName, flagValue string) error {
 	}
 	return nil
 }
+
+// ConsulLogin issues an ACL().Login to Consul and writes out the token to tokenSinkFile.
+func ConsulLogin(client *api.Client, bearerTokenFile, authMethodName, tokenSinkFile string, meta map[string]string) error {
+	data, err := ioutil.ReadFile(bearerTokenFile)
+	if err != nil {
+		return fmt.Errorf("unable to read bearerTokenFile: %v, err: %v", bearerTokenFile, err)
+	}
+	bearerToken := strings.TrimSpace(string(data))
+	if bearerToken == "" {
+		return fmt.Errorf("no bearer token found in %s", bearerTokenFile)
+	}
+	// Do the login.
+	req := &api.ACLLoginParams{
+		AuthMethod:  authMethodName, // "consul-k8s-auth-method"
+		BearerToken: bearerToken,    // /var/run/secrets/kubernetes.io/serviceaccount/token
+		Meta:        meta,           // pod=default/podName
+	}
+	tok, _, err := client.ACL().Login(req, nil)
+	if err != nil {
+		return fmt.Errorf("error logging in: %s", err)
+	}
+
+	// Write the token out to file.
+	payload := []byte(tok.SecretID)
+	if err := ioutil.WriteFile(tokenSinkFile, payload, 444); err != nil {
+		return fmt.Errorf("error writing token to file sink: %v", err)
+	}
+	return nil
+}
+
+// WriteTempFile writes contents to a temporary file and returns the file
+// name. It will remove the file once the test completes.
+func WriteTempFile(t *testing.T, contents string) string {
```

Comment:
This was moved from server-acl-init so we can re-use it.

Reply:
Note to reviewers: 
I'm moving this back and just having it be duplicated in `server-acl-init/command_test.go` and `consul-init/command_test.go` because putting it here brings in `testing` package deps to the common package.
This has the added